pandas/core/categorical.py

```diff
@@ -1085,10 +1085,14 @@ def searchsorted(self, v, side='left', sorter=None):
                              "ordered one")
 
         from pandas.core.series import Series
-        values_as_codes = self.categories.values.searchsorted(
-            Series(v).values, side=side)
+        values_as_codes = _get_codes_for_values(Series(v).values,
+                                                self.categories)
 
-        return self.codes.searchsorted(values_as_codes, sorter=sorter)
+        if -1 in values_as_codes:
```

Comment:
Sorry, could you explain it please? 
I don't see how it's possible to avoid this check and additional data scanning, if we want exception to be thrown for values not from categories list.
searchsorted() returnes 0 for -1 code (absent category), as well as for smallest category value present in the list. Thus, searchsorted() does not allow to differentiate between the value from categories which should be put on 0th position and value not from categories (0 is also returned).